File: 0242-valid-anagram/0242-valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if(s.length() != t.length()) return false;
        if(s.size() ==1 &&t.size() == 1){if(s!=t)return false;}
        int j=0;
        for(int i=0;i<s.size()-1;i++){
           if(s[i] != t[i]){
            if(i == s.size()-1)return false;
            j=i+1;
            while(s[i]!=t[i]){
                if(j==s.size()) return false;
               swap(s[i],s[j]); 
               j++;
            }
               
           }
        }
        if(s[s.size()-1]!= t[t.size()-1]) return false;
        return true;
    }
};

File: 0242-valid-anagram/0242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        if(s.length()!= t.length()) return false;
        int[] cnt = new int[256];
        for(int i=0;i<s.length();i++){
         cnt[s.charAt(i)]++;
        }
        for(int i=0;i<s.length();i++){
         if( cnt[t.charAt(i)] == 0) return false;
         cnt[t.charAt(i)]--;
        }

        for(int i=0;i<256;i++){
          if(cnt[i] < 0 || cnt[i] > 0) return false;
        }
        return true;

    }
}